orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/NoopQueue.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.q
+
+import com.netflix.spinnaker.q.DeadMessageCallback
+import com.netflix.spinnaker.q.Message
+import com.netflix.spinnaker.q.QueueCallback
+import com.netflix.spinnaker.q.metrics.EventPublisher
+import com.netflix.spinnaker.q.metrics.MonitorableQueue
+import com.netflix.spinnaker.q.metrics.NoopEventPublisher
+import com.netflix.spinnaker.q.metrics.QueueState
+import org.slf4j.LoggerFactory
+import org.threeten.extra.Minutes
+import java.time.temporal.TemporalAmount
+
+/**
+ * A Noop queue do be used when no Queue bean is found (e.g. when Queue is disabled)
+ */
+class NoopQueue : MonitorableQueue {
+  override val publisher: EventPublisher = NoopEventPublisher()
+  private val log = LoggerFactory.getLogger(this.javaClass)
+
+  init {
+    log.error("${this.javaClass.simpleName} was created - all queue operations will be NOOP'd. This is OK if the queue was intended to be disabled")
```

Comment:
it's not an error though, maybe warn instead?

Reply:
yeah, was debating back and forth with @robzienert im on the fence. The idea was to make is super obvious... what you think?

Reply:
yeah I get it, I just don't like hijacking the error log level for visibility, I'd like to keep it for actual errors. This is a legit use case for log.warn, still plenty visible IMO